daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -240,6 +240,27 @@ convertInt64 x
     | otherwise =
         unsupported "Int literal out of bounds" (negate x)
 
+convertRational :: Integer -> Integer -> ConvertM LF.Expr
+convertRational num denom
+ =
+    -- the denominator needs to be a divisor of 10^10.
+    -- num % denom * 10^10 needs to fit within a 128bit signed number.
+    -- note that we can also get negative rationals here, hence we ask for upperBound128Bit - 1 as
+    -- upper limit.
+    if | 10 ^ maxPrecision `mod` denom == 0 &&
+             abs (r * 10 ^ (10 :: Integer)) <= upperBound128Bit - 1 ->
```

Comment:
```suggestion
             abs (r * 10 ^ maxPrecision)) <= upperBound128Bit - 1 ->
```

Reply:
I wonder if this might be cleaner with `integerLog10` and potentially also faster, e.g. `integerLog10 (abs r) < 128 - maxPrecision`.

Reply:
I find the original (not using logs) clearer and closer to the semantics defined in LF.